src/Model/Todo/Todo.php

```diff
@@ -71,6 +77,24 @@ public function markAsDone()
     }
 
     /**
+     * @param \DateTimeImmutable $deadline
+     * @return void
+     * @throws \Exception
+     */
+    public function addDeadline(\DateTimeImmutable $deadline)
+    {
+        $this->recordThat(DeadlineWasAddedToTodo::byUserToDate($this->todoId, $this->assigneeId, $deadline));
```

Comment:
Before you record the event you should make sure that the deadline is valid: Should not be a date in the past!

See also comment above and move the user assertion to this location.